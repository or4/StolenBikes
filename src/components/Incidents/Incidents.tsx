import React from 'react';
import styled from 'styled-components';

import { Incident } from 'components/Incident';
import { IIncident } from 'types';

interface Props {
    incidents: IIncident[];
}

const Container = styled.div``;

export const Incidents = (props: Props) => (
    <Container>
        {props.incidents.map((incident, index) => (
            <Incident key={index} {...incident} />
        ))}
    </Container>
);
