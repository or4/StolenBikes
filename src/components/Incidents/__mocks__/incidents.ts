export const incidents = [
    {
        id: 88377,
        title: 'Stolen 2018 VAUN VELO(black)',
        description: 'Stolen from my courtyard',
        address: 'Berlin, 12047, DE',
        occurredAt: 1532152800,
        imageUrl: null,
        imageUrlThumb: null,
    },
    {
        id: 92457,
        title: 'Stolen ROCKRIDER black-grey(black and blue)',
        description: null,
        address: 'Berlin, 10999, DE',
        occurredAt: 1538931327,
        imageUrl: null,
        imageUrlThumb: null,
    },
    {
        id: 93593,
        title: 'Stolen 2018 UNIMOKE-Urban Drivestyle UNIMOKE Swing(black)',
        description:
            'someone broke into the house, cut the wooden railings beside the stairs that my bike was locked to.',
        address: 'Berlin, 10405, DE',
        occurredAt: 1540929600,
        imageUrl: 'https://files.bikeindex.org/uploads/Pu/139855/large_bike-3.jpg',
        imageUrlThumb: 'https://files.bikeindex.org/uploads/Pu/139855/small_bike-3.jpg',
        geometry: {
            lng: 13.42664,
            lat: 52.53961,
        },
    },
    {
        id: 94547,
        title: 'Stolen 2015 Gepida Alboin crs 500(silver, gray or bare metal)',
        description:
            'Locked cellar in Warthestrasse 8, Kryptonite U-lock with cable both tyres locked to frame. Someone got in and took the whole thing as well as forced some storage doors open. ',
        address: 'Berlin, 12051, DE',
        occurredAt: 1543309200,
        imageUrl: null,
        imageUrlThumb: null,
    },
    {
        id: 97530,
        title: 'Stolen 2017 Canyon bicycles Inflite AL 9.0S(green)',
        description:
            'Bike was locked to a bike rack with an "Asus Bordo Granit X-Plus" lock. After two hours in the gym, it was gone, no trace left. The rack was a bit loose in the ground, it\'s possible that the thieves tried to pull it out, but failed due to the concrete foundation.',
        address: 'Berlin, 10117, DE',
        occurredAt: 1552140000,
        imageUrl: 'https://files.bikeindex.org/uploads/Pu/148912/large_IMG_20170705_191324.jpg',
        imageUrlThumb: 'https://files.bikeindex.org/uploads/Pu/148912/small_IMG_20170705_191324.jpg',
    },
    {
        id: 97572,
        title: 'Stolen 2018 Cannondale CaadX Tiagra 58(red and black)',
        description: null,
        address: 'Berlin, 10435, DE',
        occurredAt: 1552325609,
        imageUrl: null,
        imageUrlThumb: null,
    },
    {
        id: 98451,
        title: 'Stolen 2010 Trek Superfly(black and white)',
        description: '',
        address: 'Berlin, 10787, DE',
        occurredAt: 1552554840,
        imageUrl: 'https://files.bikeindex.org/uploads/Pu/151095/large_IMG-20110221-00015.jpg',
        imageUrlThumb: 'https://files.bikeindex.org/uploads/Pu/151095/small_IMG-20110221-00015.jpg',
    },
    {
        id: 98486,
        title: 'Stolen 2019 Flyer Upstreet 4(black)',
        description: "Stolen during nighttime from the house's closed yard.",
        address: 'Berlin, 13353, DE',
        occurredAt: 1554328800,
        imageUrl: 'https://files.bikeindex.org/uploads/Pu/151183/large_upstreet-4-7-10-diamant-404900466_large.png',
        imageUrlThumb:
            'https://files.bikeindex.org/uploads/Pu/151183/small_upstreet-4-7-10-diamant-404900466_large.png',
    },
    {
        id: 98720,
        title: 'Stolen 2017 Riese und MÃ¼ller Charger GS Nuvinci(white)',
        description: null,
        address: 'Berlin, 10559, DE',
        occurredAt: 1554984394,
        imageUrl: 'https://files.bikeindex.org/uploads/Pu/151809/large_IMG_3862.jpg',
        imageUrlThumb: 'https://files.bikeindex.org/uploads/Pu/151809/small_IMG_3862.jpg',
    },
    {
        id: 99362,
        title: 'Stolen 2016 Cube Analog(blue)',
        description:
            'There were the 2 bikes I owned parked adjacent to each other and with 1 cable lock I locked the 2 bikes together and with another cable lock I locked the bike ,which was stolen, to the bike parking station bar in the courtyard of my apartment. I came back home at about 12:30 pm on 25.04.2019 and checked the bikes and found out the incident. The thief(s) cut off the cable locks and took only 1 of the bikes and left the 2nd bike.',
        address: 'Berlin, 10827, DE',
        occurredAt: 1556179200,
        imageUrl: 'https://files.bikeindex.org/uploads/Pu/153579/large_WhatsApp_Image_2019-04-25_at_14.01.49.jpeg',
        imageUrlThumb: 'https://files.bikeindex.org/uploads/Pu/153579/small_WhatsApp_Image_2019-04-25_at_14.01.49.jpeg',
    },
];
