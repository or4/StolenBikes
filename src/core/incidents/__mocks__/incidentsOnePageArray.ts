/* eslint-disable @typescript-eslint/camelcase */
export const incidentsOnePageArray = [
    {
        id: 99362,
        title: 'Stolen 2016 Cube Analog(blue)',
        description:
            'There were the 2 bikes I owned parked adjacent to each other and with 1 cable lock I locked the 2 bikes together and with another cable lock I locked the bike ,which was stolen, to the bike parking station bar in the courtyard of my apartment. I came back home at about 12:30 pm on 25.04.2019 and checked the bikes and found out the incident. The thief(s) cut off the cable locks and took only 1 of the bikes and left the 2nd bike.',
        address: 'Berlin, 10827, DE',
        occurred_at: 1556179200,
        updated_at: 1558234871,
        url: 'https://bikewise.org/api/v1/incidents/99362',
        source: {
            name: 'BikeIndex.org',
            html_url: 'https://bikeindex.org/bikes/603410',
            api_url: 'https://bikeindex.org/api/v1/bikes/603410',
        },
        media: {
            image_url: 'https://files.bikeindex.org/uploads/Pu/153579/large_WhatsApp_Image_2019-04-25_at_14.01.49.jpeg',
            image_url_thumb:
                'https://files.bikeindex.org/uploads/Pu/153579/small_WhatsApp_Image_2019-04-25_at_14.01.49.jpeg',
        },
        location_type: null,
        location_description: null,
        type: 'Theft',
        type_properties: null,
    },
    {
        id: 98720,
        title: 'Stolen 2017 Riese und MÃ¼ller Charger GS Nuvinci(white)',
        description: null,
        address: 'Berlin, 10559, DE',
        occurred_at: 1554984394,
        updated_at: 1558234893,
        url: 'https://bikewise.org/api/v1/incidents/98720',
        source: {
            name: 'BikeIndex.org',
            html_url: 'https://bikeindex.org/bikes/600502',
            api_url: 'https://bikeindex.org/api/v1/bikes/600502',
        },
        media: {
            image_url: 'https://files.bikeindex.org/uploads/Pu/151809/large_IMG_3862.jpg',
            image_url_thumb: 'https://files.bikeindex.org/uploads/Pu/151809/small_IMG_3862.jpg',
        },
        location_type: null,
        location_description: null,
        type: 'Theft',
        type_properties: null,
    },
    {
        id: 98486,
        title: 'Stolen 2019 Flyer Upstreet 4(black)',
        description: "Stolen during nighttime from the house's closed yard.",
        address: 'Berlin, 13353, DE',
        occurred_at: 1554328800,
        updated_at: 1558234904,
        url: 'https://bikewise.org/api/v1/incidents/98486',
        source: {
            name: 'BikeIndex.org',
            html_url: 'https://bikeindex.org/bikes/599290',
            api_url: 'https://bikeindex.org/api/v1/bikes/599290',
        },
        media: {
            image_url:
                'https://files.bikeindex.org/uploads/Pu/151183/large_upstreet-4-7-10-diamant-404900466_large.png',
            image_url_thumb:
                'https://files.bikeindex.org/uploads/Pu/151183/small_upstreet-4-7-10-diamant-404900466_large.png',
        },
        location_type: null,
        location_description: null,
        type: 'Theft',
        type_properties: null,
    },
    {
        id: 98451,
        title: 'Stolen 2010 Trek Superfly(black and white)',
        description: '',
        address: 'Berlin, 10787, DE',
        occurred_at: 1552554840,
        updated_at: 1558234934,
        url: 'https://bikewise.org/api/v1/incidents/98451',
        source: {
            name: 'BikeIndex.org',
            html_url: 'https://bikeindex.org/bikes/599141',
            api_url: 'https://bikeindex.org/api/v1/bikes/599141',
        },
        media: {
            image_url: 'https://files.bikeindex.org/uploads/Pu/151095/large_IMG-20110221-00015.jpg',
            image_url_thumb: 'https://files.bikeindex.org/uploads/Pu/151095/small_IMG-20110221-00015.jpg',
        },
        location_type: null,
        location_description: null,
        type: 'Theft',
        type_properties: null,
    },
    {
        id: 97572,
        title: 'Stolen 2018 Cannondale CaadX Tiagra 58(red and black)',
        description: null,
        address: 'Berlin, 10435, DE',
        occurred_at: 1552325609,
        updated_at: 1558234938,
        url: 'https://bikewise.org/api/v1/incidents/97572',
        source: {
            name: 'BikeIndex.org',
            html_url: 'https://bikeindex.org/bikes/595698',
            api_url: 'https://bikeindex.org/api/v1/bikes/595698',
        },
        media: {
            image_url: null,
            image_url_thumb: null,
        },
        location_type: null,
        location_description: null,
        type: 'Theft',
        type_properties: null,
    },
    {
        id: 97530,
        title: 'Stolen 2017 Canyon bicycles Inflite AL 9.0S(green)',
        description:
            'Bike was locked to a bike rack with an "Asus Bordo Granit X-Plus" lock. After two hours in the gym, it was gone, no trace left. The rack was a bit loose in the ground, it\'s possible that the thieves tried to pull it out, but failed due to the concrete foundation.',
        address: 'Berlin, 10117, DE',
        occurred_at: 1552140000,
        updated_at: 1558234940,
        url: 'https://bikewise.org/api/v1/incidents/97530',
        source: {
            name: 'BikeIndex.org',
            html_url: 'https://bikeindex.org/bikes/595545',
            api_url: 'https://bikeindex.org/api/v1/bikes/595545',
        },
        media: {
            image_url: 'https://files.bikeindex.org/uploads/Pu/148912/large_IMG_20170705_191324.jpg',
            image_url_thumb: 'https://files.bikeindex.org/uploads/Pu/148912/small_IMG_20170705_191324.jpg',
        },
        location_type: null,
        location_description: null,
        type: 'Theft',
        type_properties: null,
    },
    {
        id: 94547,
        title: 'Stolen 2015 Gepida Alboin crs 500(silver, gray or bare metal)',
        description:
            'Locked cellar in Warthestrasse 8, Kryptonite U-lock with cable both tyres locked to frame. Someone got in and took the whole thing as well as forced some storage doors open. ',
        address: 'Berlin, 12051, DE',
        occurred_at: 1543309200,
        updated_at: 1557813814,
        url: 'https://bikewise.org/api/v1/incidents/94547',
        source: {
            name: 'BikeIndex.org',
            html_url: 'https://bikeindex.org/bikes/480323',
            api_url: 'https://bikeindex.org/api/v1/bikes/480323',
        },
        media: {
            image_url: null,
            image_url_thumb: null,
        },
        location_type: null,
        location_description: null,
        type: 'Theft',
        type_properties: null,
    },
    {
        id: 93593,
        title: 'Stolen 2018 UNIMOKE-Urban Drivestyle UNIMOKE Swing(black)',
        description:
            'someone broke into the house, cut the wooden railings beside the stairs that my bike was locked to.',
        address: 'Berlin, 10405, DE',
        occurred_at: 1540929600,
        updated_at: 1557813775,
        url: 'https://bikewise.org/api/v1/incidents/93593',
        source: {
            name: 'BikeIndex.org',
            html_url: 'https://bikeindex.org/bikes/458285',
            api_url: 'https://bikeindex.org/api/v1/bikes/458285',
        },
        media: {
            image_url: 'https://files.bikeindex.org/uploads/Pu/139855/large_bike-3.jpg',
            image_url_thumb: 'https://files.bikeindex.org/uploads/Pu/139855/small_bike-3.jpg',
        },
        location_type: null,
        location_description: null,
        type: 'Theft',
        type_properties: null,
    },
    {
        id: 92457,
        title: 'Stolen ROCKRIDER black-grey(black and blue)',
        description: null,
        address: 'Berlin, 10999, DE',
        occurred_at: 1538931327,
        updated_at: 1557813728,
        url: 'https://bikewise.org/api/v1/incidents/92457',
        source: {
            name: 'BikeIndex.org',
            html_url: 'https://bikeindex.org/bikes/439032',
            api_url: 'https://bikeindex.org/api/v1/bikes/439032',
        },
        media: {
            image_url: null,
            image_url_thumb: null,
        },
        location_type: null,
        location_description: null,
        type: 'Theft',
        type_properties: null,
    },
    {
        id: 88377,
        title: 'Stolen 2018 VAUN VELO(black)',
        description: 'Stolen from my courtyard',
        address: 'Berlin, 12047, DE',
        occurred_at: 1532152800,
        updated_at: 1557813447,
        url: 'https://bikewise.org/api/v1/incidents/88377',
        source: {
            name: 'BikeIndex.org',
            html_url: 'https://bikeindex.org/bikes/403765',
            api_url: 'https://bikeindex.org/api/v1/bikes/403765',
        },
        media: {
            image_url: null,
            image_url_thumb: null,
        },
        location_type: null,
        location_description: null,
        type: 'Theft',
        type_properties: null,
    },
];
