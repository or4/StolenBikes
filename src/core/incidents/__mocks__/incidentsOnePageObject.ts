/* eslint-disable @typescript-eslint/camelcase */
export const incidentsOnePageObject = {
    '88377': {
        id: 88377,
        title: 'Stolen 2018 VAUN VELO(black)',
        description: 'Stolen from my courtyard',
        address: 'Berlin, 12047, DE',
        occurredAt: 1532152800,
        updatedAt: 1557813447,
        url: 'https://bikewise.org/api/v1/incidents/88377',
        source: {
            name: 'BikeIndex.org',
            htmlUrl: 'https://bikeindex.org/bikes/403765',
            apiUrl: 'https://bikeindex.org/api/v1/bikes/403765',
        },
        media: {
            imageUrl: null,
            imageUrlThumb: null,
        },
        locationType: null,
        locationDescription: null,
        type: 'Theft',
        typeProperties: null,
    },
    '92457': {
        id: 92457,
        title: 'Stolen ROCKRIDER black-grey(black and blue)',
        description: null,
        address: 'Berlin, 10999, DE',
        occurredAt: 1538931327,
        updatedAt: 1557813728,
        url: 'https://bikewise.org/api/v1/incidents/92457',
        source: {
            name: 'BikeIndex.org',
            htmlUrl: 'https://bikeindex.org/bikes/439032',
            apiUrl: 'https://bikeindex.org/api/v1/bikes/439032',
        },
        media: {
            imageUrl: null,
            imageUrlThumb: null,
        },
        locationType: null,
        locationDescription: null,
        type: 'Theft',
        typeProperties: null,
    },
    '93593': {
        id: 93593,
        title: 'Stolen 2018 UNIMOKE-Urban Drivestyle UNIMOKE Swing(black)',
        description:
            'someone broke into the house, cut the wooden railings beside the stairs that my bike was locked to.',
        address: 'Berlin, 10405, DE',
        occurredAt: 1540929600,
        updatedAt: 1557813775,
        url: 'https://bikewise.org/api/v1/incidents/93593',
        source: {
            name: 'BikeIndex.org',
            htmlUrl: 'https://bikeindex.org/bikes/458285',
            apiUrl: 'https://bikeindex.org/api/v1/bikes/458285',
        },
        media: {
            imageUrl: 'https://files.bikeindex.org/uploads/Pu/139855/large_bike-3.jpg',
            imageUrlThumb: 'https://files.bikeindex.org/uploads/Pu/139855/small_bike-3.jpg',
        },
        locationType: null,
        locationDescription: null,
        type: 'Theft',
        typeProperties: null,
    },
    '94547': {
        id: 94547,
        title: 'Stolen 2015 Gepida Alboin crs 500(silver, gray or bare metal)',
        description:
            'Locked cellar in Warthestrasse 8, Kryptonite U-lock with cable both tyres locked to frame. Someone got in and took the whole thing as well as forced some storage doors open. ',
        address: 'Berlin, 12051, DE',
        occurredAt: 1543309200,
        updatedAt: 1557813814,
        url: 'https://bikewise.org/api/v1/incidents/94547',
        source: {
            name: 'BikeIndex.org',
            htmlUrl: 'https://bikeindex.org/bikes/480323',
            apiUrl: 'https://bikeindex.org/api/v1/bikes/480323',
        },
        media: {
            imageUrl: null,
            imageUrlThumb: null,
        },
        locationType: null,
        locationDescription: null,
        type: 'Theft',
        typeProperties: null,
    },
    '97530': {
        id: 97530,
        title: 'Stolen 2017 Canyon bicycles Inflite AL 9.0S(green)',
        description:
            'Bike was locked to a bike rack with an "Asus Bordo Granit X-Plus" lock. After two hours in the gym, it was gone, no trace left. The rack was a bit loose in the ground, it\'s possible that the thieves tried to pull it out, but failed due to the concrete foundation.',
        address: 'Berlin, 10117, DE',
        occurredAt: 1552140000,
        updatedAt: 1558234940,
        url: 'https://bikewise.org/api/v1/incidents/97530',
        source: {
            name: 'BikeIndex.org',
            htmlUrl: 'https://bikeindex.org/bikes/595545',
            apiUrl: 'https://bikeindex.org/api/v1/bikes/595545',
        },
        media: {
            imageUrl: 'https://files.bikeindex.org/uploads/Pu/148912/large_IMG_20170705_191324.jpg',
            imageUrlThumb: 'https://files.bikeindex.org/uploads/Pu/148912/small_IMG_20170705_191324.jpg',
        },
        locationType: null,
        locationDescription: null,
        type: 'Theft',
        typeProperties: null,
    },
    '97572': {
        id: 97572,
        title: 'Stolen 2018 Cannondale CaadX Tiagra 58(red and black)',
        description: null,
        address: 'Berlin, 10435, DE',
        occurredAt: 1552325609,
        updatedAt: 1558234938,
        url: 'https://bikewise.org/api/v1/incidents/97572',
        source: {
            name: 'BikeIndex.org',
            htmlUrl: 'https://bikeindex.org/bikes/595698',
            apiUrl: 'https://bikeindex.org/api/v1/bikes/595698',
        },
        media: {
            imageUrl: null,
            imageUrlThumb: null,
        },
        locationType: null,
        locationDescription: null,
        type: 'Theft',
        typeProperties: null,
    },
    '98451': {
        id: 98451,
        title: 'Stolen 2010 Trek Superfly(black and white)',
        description: '',
        address: 'Berlin, 10787, DE',
        occurredAt: 1552554840,
        updatedAt: 1558234934,
        url: 'https://bikewise.org/api/v1/incidents/98451',
        source: {
            name: 'BikeIndex.org',
            htmlUrl: 'https://bikeindex.org/bikes/599141',
            apiUrl: 'https://bikeindex.org/api/v1/bikes/599141',
        },
        media: {
            imageUrl: 'https://files.bikeindex.org/uploads/Pu/151095/large_IMG-20110221-00015.jpg',
            imageUrlThumb: 'https://files.bikeindex.org/uploads/Pu/151095/small_IMG-20110221-00015.jpg',
        },
        locationType: null,
        locationDescription: null,
        type: 'Theft',
        typeProperties: null,
    },
    '98486': {
        id: 98486,
        title: 'Stolen 2019 Flyer Upstreet 4(black)',
        description: "Stolen during nighttime from the house's closed yard.",
        address: 'Berlin, 13353, DE',
        occurredAt: 1554328800,
        updatedAt: 1558234904,
        url: 'https://bikewise.org/api/v1/incidents/98486',
        source: {
            name: 'BikeIndex.org',
            htmlUrl: 'https://bikeindex.org/bikes/599290',
            apiUrl: 'https://bikeindex.org/api/v1/bikes/599290',
        },
        media: {
            imageUrl: 'https://files.bikeindex.org/uploads/Pu/151183/large_upstreet-4-7-10-diamant-404900466_large.png',
            imageUrlThumb:
                'https://files.bikeindex.org/uploads/Pu/151183/small_upstreet-4-7-10-diamant-404900466_large.png',
        },
        locationType: null,
        locationDescription: null,
        type: 'Theft',
        typeProperties: null,
    },
    '98720': {
        id: 98720,
        title: 'Stolen 2017 Riese und MÃ¼ller Charger GS Nuvinci(white)',
        description: null,
        address: 'Berlin, 10559, DE',
        occurredAt: 1554984394,
        updatedAt: 1558234893,
        url: 'https://bikewise.org/api/v1/incidents/98720',
        source: {
            name: 'BikeIndex.org',
            htmlUrl: 'https://bikeindex.org/bikes/600502',
            apiUrl: 'https://bikeindex.org/api/v1/bikes/600502',
        },
        media: {
            imageUrl: 'https://files.bikeindex.org/uploads/Pu/151809/large_IMG_3862.jpg',
            imageUrlThumb: 'https://files.bikeindex.org/uploads/Pu/151809/small_IMG_3862.jpg',
        },
        locationType: null,
        locationDescription: null,
        type: 'Theft',
        typeProperties: null,
    },
    '99362': {
        id: 99362,
        title: 'Stolen 2016 Cube Analog(blue)',
        description:
            'There were the 2 bikes I owned parked adjacent to each other and with 1 cable lock I locked the 2 bikes together and with another cable lock I locked the bike ,which was stolen, to the bike parking station bar in the courtyard of my apartment. I came back home at about 12:30 pm on 25.04.2019 and checked the bikes and found out the incident. The thief(s) cut off the cable locks and took only 1 of the bikes and left the 2nd bike.',
        address: 'Berlin, 10827, DE',
        occurredAt: 1556179200,
        updatedAt: 1558234871,
        url: 'https://bikewise.org/api/v1/incidents/99362',
        source: {
            name: 'BikeIndex.org',
            htmlUrl: 'https://bikeindex.org/bikes/603410',
            apiUrl: 'https://bikeindex.org/api/v1/bikes/603410',
        },
        media: {
            imageUrl: 'https://files.bikeindex.org/uploads/Pu/153579/large_WhatsApp_Image_2019-04-25_at_14.01.49.jpeg',
            imageUrlThumb:
                'https://files.bikeindex.org/uploads/Pu/153579/small_WhatsApp_Image_2019-04-25_at_14.01.49.jpeg',
        },
        locationType: null,
        locationDescription: null,
        type: 'Theft',
        typeProperties: null,
    },
};
