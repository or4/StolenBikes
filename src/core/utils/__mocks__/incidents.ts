/* eslint-disable @typescript-eslint/camelcase */
export const incidentsCamelCaseArray = [
    {
        address: 'Berlin, 10827, DE',
        description:
            'There were the 2 bikes I owned parked adjacent to each other and with 1 cable lock I locked the 2 bikes together and with another cable lock I locked the bike ,which was stolen, to the bike parking station bar in the courtyard of my apartment. I came back home at about 12:30 pm on 25.04.2019 and checked the bikes and found out the incident. The thief(s) cut off the cable locks and took only 1 of the bikes and left the 2nd bike.',
        id: 99362,
        locationDescription: null,
        locationType: null,
        media: {
            imageUrl: 'https://files.bikeindex.org/uploads/Pu/153579/large_WhatsApp_Image_2019-04-25_at_14.01.49.jpeg',
            imageUrlThumb:
                'https://files.bikeindex.org/uploads/Pu/153579/small_WhatsApp_Image_2019-04-25_at_14.01.49.jpeg',
        },
        occurredAt: 1556179200,
        source: {
            name: 'BikeIndex.org',
            htmlUrl: 'https://bikeindex.org/bikes/603410',
            apiUrl: 'https://bikeindex.org/api/v1/bikes/603410',
        },
        title: 'Stolen 2016 Cube Analog(blue)',
        type: 'Theft',
        typeProperties: null,
        updatedAt: 1558180870,
        url: 'https://bikewise.org/api/v1/incidents/99362',
    },
    {
        id: 98720,
        title: 'Stolen 2017 Riese und Müller Charger GS Nuvinci(white)',
        description: null,
        address: 'Berlin, 10559, DE',
        occurredAt: 1554984394,
        updatedAt: 1558234893,
        url: 'https://bikewise.org/api/v1/incidents/98720',
        source: {
            name: 'BikeIndex.org',
            htmlUrl: 'https://bikeindex.org/bikes/600502',
            apiUrl: 'https://bikeindex.org/api/v1/bikes/600502',
        },
        media: {
            imageUrl: 'https://files.bikeindex.org/uploads/Pu/151809/large_IMG_3862.jpg',
            imageUrlThumb: 'https://files.bikeindex.org/uploads/Pu/151809/small_IMG_3862.jpg',
        },
        locationType: null,
        locationDescription: null,
        type: 'Theft',
        typeProperties: null,
    },
];

export const incidentsSnakeCaseArray = [
    {
        address: 'Berlin, 10827, DE',
        description:
            'There were the 2 bikes I owned parked adjacent to each other and with 1 cable lock I locked the 2 bikes together and with another cable lock I locked the bike ,which was stolen, to the bike parking station bar in the courtyard of my apartment. I came back home at about 12:30 pm on 25.04.2019 and checked the bikes and found out the incident. The thief(s) cut off the cable locks and took only 1 of the bikes and left the 2nd bike.',
        id: 99362,
        location_description: null,
        location_type: null,
        media: {
            image_url: 'https://files.bikeindex.org/uploads/Pu/153579/large_WhatsApp_Image_2019-04-25_at_14.01.49.jpeg',
            image_url_thumb:
                'https://files.bikeindex.org/uploads/Pu/153579/small_WhatsApp_Image_2019-04-25_at_14.01.49.jpeg',
        },
        occurred_at: 1556179200,
        source: {
            name: 'BikeIndex.org',
            html_url: 'https://bikeindex.org/bikes/603410',
            api_url: 'https://bikeindex.org/api/v1/bikes/603410',
        },
        title: 'Stolen 2016 Cube Analog(blue)',
        type: 'Theft',
        type_properties: null,
        updated_at: 1558180870,
        url: 'https://bikewise.org/api/v1/incidents/99362',
    },
    {
        id: 98720,
        title: 'Stolen 2017 Riese und Müller Charger GS Nuvinci(white)',
        description: null,
        address: 'Berlin, 10559, DE',
        occurred_at: 1554984394,
        updated_at: 1558234893,
        url: 'https://bikewise.org/api/v1/incidents/98720',
        source: {
            name: 'BikeIndex.org',
            html_url: 'https://bikeindex.org/bikes/600502',
            api_url: 'https://bikeindex.org/api/v1/bikes/600502',
        },
        media: {
            image_url: 'https://files.bikeindex.org/uploads/Pu/151809/large_IMG_3862.jpg',
            image_url_thumb: 'https://files.bikeindex.org/uploads/Pu/151809/small_IMG_3862.jpg',
        },
        location_type: null,
        location_description: null,
        type: 'Theft',
        type_properties: null,
    },
];
