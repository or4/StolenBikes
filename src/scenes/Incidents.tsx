import React from 'react';
import { Header } from 'components/Header/Header';

export class Incidents extends React.Component {
    public render(): React.ReactElement {
        return <Header />;
    }
}
