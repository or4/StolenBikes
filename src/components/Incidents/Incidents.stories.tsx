import * as React from 'react';
import { MemoryRouter } from 'react-router';
import { storiesOf } from '@storybook/react';

import { Incidents } from './Incidents';
import { incidents } from './__mocks__/incidents';

const stories = storiesOf('Incidents', module);
stories.addDecorator(story => <MemoryRouter initialEntries={['/']}>{story()}</MemoryRouter>);

stories.add('Incidents', () => {
    return <Incidents incidents={incidents} />;
});
