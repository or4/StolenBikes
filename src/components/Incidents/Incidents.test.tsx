import React from 'react';
import { shallow } from 'enzyme';

import { Incidents } from './Incidents';
import { incidents } from './__mocks__/incidents';

describe('Incidents', () => {
    it('should render correctly', () => {
        const component = <Incidents incidents={incidents} />;

        const container = shallow(component);
        expect(container).toMatchSnapshot();
        container.unmount();
    });
});
