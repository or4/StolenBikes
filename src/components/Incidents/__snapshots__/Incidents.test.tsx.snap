// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Incidents should render correctly 1`] = `
<styled.div>
  <Incident
    address="Berlin, 12047, DE"
    description="Stolen from my courtyard"
    id={88377}
    imageUrl={null}
    imageUrlThumb={null}
    key="0"
    occurredAt={1532152800}
    title="Stolen 2018 VAUN VELO(black)"
  />
  <Incident
    address="Berlin, 10999, DE"
    description={null}
    id={92457}
    imageUrl={null}
    imageUrlThumb={null}
    key="1"
    occurredAt={1538931327}
    title="Stolen ROCKRIDER black-grey(black and blue)"
  />
  <Incident
    address="Berlin, 10405, DE"
    description="someone broke into the house, cut the wooden railings beside the stairs that my bike was locked to."
    geometry={
      Object {
        "lat": 52.53961,
        "lng": 13.42664,
      }
    }
    id={93593}
    imageUrl="https://files.bikeindex.org/uploads/Pu/139855/large_bike-3.jpg"
    imageUrlThumb="https://files.bikeindex.org/uploads/Pu/139855/small_bike-3.jpg"
    key="2"
    occurredAt={1540929600}
    title="Stolen 2018 UNIMOKE-Urban Drivestyle UNIMOKE Swing(black)"
  />
  <Incident
    address="Berlin, 12051, DE"
    description="Locked cellar in Warthestrasse 8, Kryptonite U-lock with cable both tyres locked to frame. Someone got in and took the whole thing as well as forced some storage doors open. "
    id={94547}
    imageUrl={null}
    imageUrlThumb={null}
    key="3"
    occurredAt={1543309200}
    title="Stolen 2015 Gepida Alboin crs 500(silver, gray or bare metal)"
  />
  <Incident
    address="Berlin, 10117, DE"
    description="Bike was locked to a bike rack with an \\"Asus Bordo Granit X-Plus\\" lock. After two hours in the gym, it was gone, no trace left. The rack was a bit loose in the ground, it's possible that the thieves tried to pull it out, but failed due to the concrete foundation."
    id={97530}
    imageUrl="https://files.bikeindex.org/uploads/Pu/148912/large_IMG_20170705_191324.jpg"
    imageUrlThumb="https://files.bikeindex.org/uploads/Pu/148912/small_IMG_20170705_191324.jpg"
    key="4"
    occurredAt={1552140000}
    title="Stolen 2017 Canyon bicycles Inflite AL 9.0S(green)"
  />
  <Incident
    address="Berlin, 10435, DE"
    description={null}
    id={97572}
    imageUrl={null}
    imageUrlThumb={null}
    key="5"
    occurredAt={1552325609}
    title="Stolen 2018 Cannondale CaadX Tiagra 58(red and black)"
  />
  <Incident
    address="Berlin, 10787, DE"
    description=""
    id={98451}
    imageUrl="https://files.bikeindex.org/uploads/Pu/151095/large_IMG-20110221-00015.jpg"
    imageUrlThumb="https://files.bikeindex.org/uploads/Pu/151095/small_IMG-20110221-00015.jpg"
    key="6"
    occurredAt={1552554840}
    title="Stolen 2010 Trek Superfly(black and white)"
  />
  <Incident
    address="Berlin, 13353, DE"
    description="Stolen during nighttime from the house's closed yard."
    id={98486}
    imageUrl="https://files.bikeindex.org/uploads/Pu/151183/large_upstreet-4-7-10-diamant-404900466_large.png"
    imageUrlThumb="https://files.bikeindex.org/uploads/Pu/151183/small_upstreet-4-7-10-diamant-404900466_large.png"
    key="7"
    occurredAt={1554328800}
    title="Stolen 2019 Flyer Upstreet 4(black)"
  />
  <Incident
    address="Berlin, 10559, DE"
    description={null}
    id={98720}
    imageUrl="https://files.bikeindex.org/uploads/Pu/151809/large_IMG_3862.jpg"
    imageUrlThumb="https://files.bikeindex.org/uploads/Pu/151809/small_IMG_3862.jpg"
    key="8"
    occurredAt={1554984394}
    title="Stolen 2017 Riese und MÃ¼ller Charger GS Nuvinci(white)"
  />
  <Incident
    address="Berlin, 10827, DE"
    description="There were the 2 bikes I owned parked adjacent to each other and with 1 cable lock I locked the 2 bikes together and with another cable lock I locked the bike ,which was stolen, to the bike parking station bar in the courtyard of my apartment. I came back home at about 12:30 pm on 25.04.2019 and checked the bikes and found out the incident. The thief(s) cut off the cable locks and took only 1 of the bikes and left the 2nd bike."
    id={99362}
    imageUrl="https://files.bikeindex.org/uploads/Pu/153579/large_WhatsApp_Image_2019-04-25_at_14.01.49.jpeg"
    imageUrlThumb="https://files.bikeindex.org/uploads/Pu/153579/small_WhatsApp_Image_2019-04-25_at_14.01.49.jpeg"
    key="9"
    occurredAt={1556179200}
    title="Stolen 2016 Cube Analog(blue)"
  />
</styled.div>
`;
